odoo/models.py

```diff
@@ -2460,6 +2475,123 @@ def _read_group_format_result(self, rows_dict, lazy_groupby):
+            for row in rows_dict:
+                if not row[fullname]:
+                    # can not only do ('many2one', '=', False) because we might have
+                    # record in database that does not exist anymore
+                    additional_domain = expression.OR([
+                        [(fullname, '=', False)],
+                        [(fullname, 'not in', all_groups)],
+                    ])
+                else:
+                    additional_domain = [(fullname, '=', row[fullname])]
+                    record = self.env[comodel].browse(row[fullname]).with_prefetch(prefetch_ids)
+                    row[fullname] = (row[fullname], record.display_name)
+
+                row['__domain'] = expression.AND([row['__domain'], additional_domain])
+
+        elif property_type == 'many2many':
+            comodel = definition.get('comodel')
+            prefetch_ids = tuple(row[fullname] for row in rows_dict if row[fullname])
+            all_groups = tuple(row[fullname] for row in rows_dict if row[fullname])
+            for row in rows_dict:
+                if not row[fullname]:
+                    additional_domain = expression.OR([
+                        [(fullname, '=', False)],
+                        [(fullname, 'not in', group) for group in all_groups],
+                    ]) if all_groups else []
+                else:
+                    additional_domain = [(fullname, 'in', row[fullname])]
+                    record = self.env[comodel].browse(row[fullname]).with_prefetch(prefetch_ids)
+                    row[fullname] = (row[fullname], record.display_name)
+
+                row['__domain'] = expression.AND([row['__domain'], additional_domain])
+
+        elif property_type == 'tags':
+            tags = definition.get('tags') or []
+            tags = {tag[0]: tag for tag in tags}
+            for row in rows_dict:
+                if not row[fullname]:
+                    additional_domain = expression.OR([
+                        [(fullname, '=', False)],
+                        [(fullname, 'not in', tag) for tag in tags],
+                    ]) if tags else []
+                else:
+                    additional_domain = [(fullname, 'in', row[fullname])]
+                    # replace tag raw value with list of raw value, label and color
+                    row[fullname] = tags.get(row[fullname])
+
+                row['__domain'] = expression.AND([row['__domain'], additional_domain])
+
+        elif property_type in ('date', 'datetime'):
+            for row in rows_dict:
+                if not row[group]:
+                    row[group] = False
+                    row['__domain'] = expression.AND([row['__domain'], [(fullname, '=', False)]])
+                    row['__range'] = {}
+                    continue
+
+                # Date / Datetime are not JSONifiable, so they are stored as raw text
+                db_format = '%Y-%m-%d' if property_type == 'date' else '%Y-%m-%d %H:%M:%S'
```

Comment:
Shouldn't we use DEFAULT_SERVER_DATE_FORMAT and DEFAULT_SERVER_DATETIME_FORMAT ?